tests/plugins/test_check_base_image.py

```diff
@@ -77,100 +72,81 @@ def test_pull_base_image_special(add_another_parent, special_image, change_base,
-
-    (None, BASE_IMAGE,
-     # expected:
-     [],
-     # not expected:
-     [BASE_IMAGE_W_REGISTRY, BASE_IMAGE_W_LIB_REG]),
-
-    (None, BASE_IMAGE_W_REGISTRY,
-     # expected:
-     [BASE_IMAGE_W_REGISTRY],
-     # not expected:
-     [BASE_IMAGE_W_LIB_REG]),
-
-    # Tests with explicit "library" namespace:
-
-    (LOCALHOST_REGISTRY, BASE_IMAGE_W_LIB_REG,
-     # expected:
-     [],
-     # not expected:
-     [BASE_IMAGE_W_REGISTRY]),
-
-    (None, BASE_IMAGE_W_LIB_REG,
-     # expected:
-     [],
-     # not expected:
-     [BASE_IMAGE_W_REGISTRY]),
-])
-def test_pull_base_image_plugin(workflow, parent_registry, df_base, expected, not_expected,
-                                inspect_only, workflow_callback=None,
-                                check_platforms=False, parent_images=None, organization=None,
-                                parent_images_digests=None, expected_digests=None,
-                                pull_registries=None):
+                             (LOCALHOST_REGISTRY, BASE_IMAGE,
+                              # expected:
+                              [BASE_IMAGE_W_REGISTRY],
+                              # not expected:
+                              [BASE_IMAGE_W_LIB_REG]),
+
+                             (LOCALHOST_REGISTRY, BASE_IMAGE_W_REGISTRY,
+                              # expected:
+                              [BASE_IMAGE_W_REGISTRY],
+                              # not expected:
+                              [BASE_IMAGE_W_LIB_REG]),
+
+                             (None, BASE_IMAGE,
+                              # expected:
+                              [],
+                              # not expected:
+                              [BASE_IMAGE_W_REGISTRY, BASE_IMAGE_W_LIB_REG]),
+
+                             (None, BASE_IMAGE_W_REGISTRY,
+                              # expected:
+                              [BASE_IMAGE_W_REGISTRY],
+                              # not expected:
+                              [BASE_IMAGE_W_LIB_REG]),
+
+                             # Tests with explicit "library" namespace:
+
+                             (LOCALHOST_REGISTRY, BASE_IMAGE_W_LIB_REG,
+                              # expected:
+                              [],
+                              # not expected:
+                              [BASE_IMAGE_W_REGISTRY]),
+
+                             (None, BASE_IMAGE_W_LIB_REG,
+                              # expected:
+                              [],
+                              # not expected:
+                              [BASE_IMAGE_W_REGISTRY]),
+                         ])
+def test_check_base_image_plugin(workflow, parent_registry, df_base, expected, not_expected,
+                                 workflow_callback=None, parent_images=None, organization=None,
+                                 expected_digests=None, pull_registries=None,
+                                 mock_get_manifest_list=True):
```

Comment:
It appears that both `expected` and `not_expected` are essentially unused now.

I did not find any test that uses `expected_digests` so we might be missing some coverage, but it seems that that was already the case before.

I also did not find any test that checks if `workflow.dockerfile_images` actually contains manifest list digests after the plugin runs. Do we have one?

Reply:
removed `expected` and `not_expected` parameters. Added a used for` expected_digests`. Added a check to ensure `dockerfile_images` contains `sha256` digests.